cpp-client/deephaven/client/build-dependencies.sh

```diff
@@ -4,114 +4,183 @@
+NCPUS=$(getconf _NPROCESSORS_ONLN)
+
+# Where the checked out sources for dependencies will go
+export SRC=$DHDEPS_HOME/src
+
+# Where the install prefix paths will go
+export PFX=$DHDEPS_HOME/local
+
+# Let's get make to print out commands as they run
+export VERBOSE=1
+
 export CMAKE_PREFIX_PATH=${PFX}/abseil:${PFX}/cares:${PFX}/flatbuffers:${PFX}/gflags:${PFX}/protobuf:${PFX}/re2:${PFX}/zlib:${PFX}/grpc:${PFX}/arrow:${PFX}/deephaven
 
+# If you want to rebuild only certain parts or skip phases change "yes" to "no" below.
+: ${CHECKOUT:=yes}
+: ${BUILD_PROTOBUF:=yes}
+: ${BUILD_RE2:=yes}
+: ${BUILD_GFLAGS:=yes}
+: ${BUILD_ABSL:=yes}
+: ${BUILD_FLATBUFFERS:=yes}
+: ${BUILD_CARES:=yes}
+: ${BUILD_ZLIB:=yes}
+: ${BUILD_GRPC:=yes}
+: ${BUILD_ARROW:=yes}
+
+if [ ! -d $SRC ]; then
+  mkdir -p $SRC
+fi
+
+if [ ! -d $PFX ]; then
+  mkdir -p $PFX
+fi
 
-cd $MYSRC
-git clone -b v3.18.0 --depth 1 https://github.com/protocolbuffers/protobuf.git
-git clone -b 2021-09-01 --depth 1 https://github.com/google/re2.git
-git clone -b v2.2.2 --depth 1 https://github.com/gflags/gflags.git
-git clone -b 20210324.2 --depth 1 https://github.com/abseil/abseil-cpp.git
-git clone -b v2.0.0 --depth 1 https://github.com/google/flatbuffers.git
-git clone -b cares-1_17_2 --depth 1 https://github.com/c-ares/c-ares.git
-git clone -b v1.2.11 --depth 1 https://github.com/madler/zlib
-git clone -b v1.38.0 --depth 1 https://github.com/grpc/grpc
-wget 'https://www.apache.org/dyn/closer.lua?action=download&filename=arrow/arrow-5.0.0/apache-arrow-5.0.0.tar.gz' -O apache-arrow-5.0.0.tar.gz
-tar xfz apache-arrow-5.0.0.tar.gz
-mkdir -p ${PFX}
+#
+# Each phase below should explicitly change to expected current working directory before starting;
+# there is no guarantee where the CWD is after a prior phase.
+#
+
+if [ "$CHECKOUT" = "yes" ]; then
+  cd $SRC
+  git clone -b v3.18.0 --depth 1 https://github.com/protocolbuffers/protobuf.git
+  git clone -b 2021-09-01 --depth 1 https://github.com/google/re2.git
+  git clone -b v2.2.2 --depth 1 https://github.com/gflags/gflags.git
+  git clone -b 20210324.2 --depth 1 https://github.com/abseil/abseil-cpp.git
+  git clone -b v2.0.0 --depth 1 https://github.com/google/flatbuffers.git
+  git clone -b cares-1_17_2 --depth 1 https://github.com/c-ares/c-ares.git
+  git clone -b v1.2.11 --depth 1 https://github.com/madler/zlib
+  git clone -b v1.38.0 --depth 1 https://github.com/grpc/grpc
+  wget 'https://www.apache.org/dyn/closer.lua?action=download&filename=arrow/arrow-5.0.0/apache-arrow-5.0.0.tar.gz' -O apache-arrow-5.0.0.tar.gz
+  tar xfz apache-arrow-5.0.0.tar.gz
+  rm -f apache-arrow-5.0.0.tar.gz
+fi 
 
 ### Protobuf
-echo
-echo "*** Building protobuf"
-cd $MYSRC/protobuf
-mkdir -p cmake/build && cd cmake/build
-cmake -Dprotobuf_BUILD_TESTS=OFF -DCMAKE_POSITION_INDEPENDENT_CODE=TRUE -DCMAKE_BUILD_TYPE=${BUILD_TYPE} -DCMAKE_INSTALL_PREFIX=${PFX}/protobuf ..
-make -j$NCPUS
-make install
+if [ "$BUILD_PROTOBUF" = "yes" ]; then
```

Comment:
I don't think the "x$Y" idiom is really needed in this context, because the variables are quoted.  If they weren't it would make a difference, but with a quote the expansion is the empty string (as opposed to {nothing}).

https://stackoverflow.com/questions/174119/why-do-shell-script-comparisons-often-use-xvar-xyes